upload/extension/opencart/admin/controller/payment/opayo.php

```diff
@@ -0,0 +1,351 @@
+
+		$data['geo_zones'] = $this->model_localisation_geo_zone->getGeoZones();
+
+		$data['status'] = $this->config->get('payment_opayo_status');
+		$data['sort_order'] = $this->config->get('payment_opayo_sort_order');
+
+		if (!$data['setting']['cron']['token']) {
+			$data['setting']['cron']['token'] = sha1(uniqid(mt_rand(), 1));
+		}
+
+		if (!$data['setting']['cron']['url']) {
+			$data['setting']['cron']['url'] = $catalog . 'index.php?route=extension/opayo/payment' . $this->separator . 'cron&token=' . $data['setting']['cron']['token'];
+		}
+
+		$data['header'] = $this->load->controller('common/header');
+		$data['column_left'] = $this->load->controller('common/column_left');
+		$data['footer'] = $this->load->controller('common/footer');
+
+		$this->response->setOutput($this->load->view('extension/opayo/payment/opayo', $data));
+	}
+
+	public function save(): void {
+		$this->load->language('extension/opayo/payment/opayo');
+
+		if (($this->request->server['REQUEST_METHOD'] == 'POST') && $this->validate()) {
+			$this->load->model('setting/setting');
+
+			$this->model_setting_setting->editSetting('payment_opayo', $this->request->post);
+
+			$data['success'] = $this->language->get('success_save');
+		}
+
+		$data['error'] = $this->error;
+
+		$this->response->addHeader('Content-Type: application/json');
+		$this->response->setOutput(json_encode($data));
+	}
+
+	/**
+	 * Install
+	 *
+	 * @return void
+	 */
+	public function install(): void {
+		$this->load->model('extension/opayo/payment/opayo');
+
+		$this->model_extension_opayo_payment_opayo->install();
+	}
+
+	/**
+	 * Uninstall
+	 *
+	 * @return void
+	 */
+	public function uninstall(): void {
+		$this->load->model('extension/opayo/payment/opayo');
+
+		$this->model_extension_opayo_payment_opayo->uninstall();
+	}
+
+	/**
+	 * Order
+	 *
+	 * @return string
+	 */
+	public function order(): string {
+		if ($this->config->get('payment_opayo_status')) {
+			$this->load->model('extension/opayo/payment/opayo');
+
+			$opayo_order = $this->model_extension_opayo_payment_opayo->getOrder($this->request->get['order_id']);
+
+			if (!empty($opayo_order)) {
+				$this->load->language('extension/opayo/payment/opayo');
+
+				$opayo_order['total_released'] = $this->model_extension_opayo_payment_opayo->getTotalReleased($opayo_order['opayo_order_id']);
+
+				$opayo_order['total_formatted'] = $this->currency->format($opayo_order['total'], $opayo_order['currency_code'], false, false);
+				$opayo_order['total_released_formatted'] = $this->currency->format($opayo_order['total_released'], $opayo_order['currency_code'], false, false);
+
+				$data['opayo_order'] = $opayo_order;
+				$data['auto_settle'] = $opayo_order['settle_type'];
+
+				$data['order_id'] = $this->request->get['order_id'];
+
+				$data['user_token'] = $this->request->get['user_token'];
+
+				$data['separator'] = $this->separator;
+
+				return $this->load->view('extension/opayo/payment/order', $data);
+			}
+		} else {
+			return '';
+		}
```

Comment:
This will make the function also return an empty string if $opayo_order is empty, which fixes the missing return statment.
```suggestion
		}

		return '';
```